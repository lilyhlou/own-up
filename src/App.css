.App {
  text-align: center;
}
.input {
  display: flex;
  white-space:nowrap
}
.formBox {
  min-width: 18vw;
  margin-left: 2rem;
  margin-right:10rem;
}
.dropdown-menu{
  width: 100%;
  text-align: center;
}
form {
  margin-top: 10vh;
}
.button {
  margin-right: 6vw;
}
.blackBackground {
  background-color:#143f3d!important;
  border-color: #143f3d!important;
}
body {
  color: #143f3d;
}

@media only screen and (max-width: 1080px) {
  #forMedScreen .justify-content-end.d-flex.button {
    margin-right: 0rem;
  }

  .formBox {
    min-width: 24vw;
    margin-left: 1vw;
    margin-right:0rem;
  }
.form-inline label {
  display: block
}
.col-md {
  padding-right: 10vw!important;
}
}

@media only screen and (max-width: 800px) {
  #forMedScreen .justify-content-end.d-flex {
    justify-content: center!important;
  }
  #forMedScreen .justify-content-end.d-flex.button {
    margin-right: 0rem;
  }
  .form-row>.col {
    padding-left: 0px;
  }
  .col-md {
    padding-right: 0px!important;
  }
  .formBox {
    margin-left: .5vw;
    margin-right:0rem;
  }
}

@media only screen and (min-width: 1300px) {
  .formBox {
    min-width: 18vw;
    margin-left: 1vw;
    margin-right:14vw;
  }
  .btn {
    margin-right: 8vw;
  }
}
@media only screen and (max-width: 574px) {
  .formBox {
    margin-left: .5vw;
    min-width: 45vw;
  }
}