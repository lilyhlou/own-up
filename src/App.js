import './App.css';
import InputForm from './components/Input';
import Table from './components/Table'
import configureStore from './store/configureStore';
import { Provider } from 'react-redux';
import ReactDOM, { render } from 'react-dom';

function App() {
  const store = configureStore(); // You can also pass in an initialState here

  return (
    <Provider store={store}>
    <div className="App">
        <InputForm></InputForm>
        <Table></Table>
    </div>
    </Provider>
  );
}

export default App;
