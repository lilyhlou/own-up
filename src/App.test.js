import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import App from './App';
import ReactDOM from 'react-dom';


const mockStore = configureStore([]);
 
describe('App.js snapshot test', () => {
  let store;
  let component;
 
  beforeEach(() => {
      store = mockStore({
      items: [],
      itemsHasErrored: false,
      itemsIsLoading: false,
    });
    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });
 
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <App />
      </Provider>
    , div);
    ReactDOM.unmountComponentAtNode(div);

  })
 });

