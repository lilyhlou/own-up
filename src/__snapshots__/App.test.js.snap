// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="App"
>
  <div
    className="container-fluid"
  >
    <form
      className=""
      id="forMedScreen"
    >
      <div
        className="form-row"
      >
        <div
          className="d-flex justify-content-end form-group col-lg-2 col-md-2"
        >
          <label
            className="form-label"
          >
            Loan Size
          </label>
        </div>
        <div
          className="d-flex justify-content-start form-group col-lg-4 col-md-4"
        >
          <div
            className="formBox mb-3 noerror input-group"
          >
            <div
              className="input-group-prepend"
            >
              <span
                className="input-group-text"
              >
                $
              </span>
            </div>
            <input
              className="form-control form-control-md"
              min="300"
              name="loan"
              onChange={[Function]}
              placeholder="450,000"
              type="number"
            />
          </div>
        </div>
        <div
          className="d-flex justify-content-end form-group col-lg-2 col-md-2"
        >
          <label
            className="form-label"
          >
            Property Type
          </label>
        </div>
        <div
          className="d-flex justify-content-start form-group col-lg-4 col-md-4"
        >
          <select
            className="formBox noerror form-control form-control-md"
            defaultValue="DEFAULT"
            name="property"
            onChange={[Function]}
          >
            <option
              disabled={true}
              hidden={true}
              value="DEFAULT"
            >
              Select Option
            </option>
            <option
              value="SingleFamily"
            >
              Single Family
            </option>
            <option
              value="Condo"
            >
              Condo
            </option>
            <option
              value="Townhouse"
            >
              Townhouse
            </option>
            <option
              value="MultiFamily"
            >
              Multi-Family
            </option>
          </select>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="d-flex justify-content-end form-group col-md-2"
        >
          <label
            className="form-label"
          >
            Credit Score
          </label>
        </div>
        <div
          className="d-flex justify-content-start form-group col-lg-4 col-md-4"
        >
          <input
            className="formBox noerror form-control form-control-md"
            max="800"
            min="300"
            name="credit"
            onChange={[Function]}
            placeholder="300-800"
            type="number"
          />
        </div>
        <div
          className="d-flex justify-content-end form-group col-lg-2 col-md-2"
        >
          <label
            className="form-label"
          >
            Occupancy
          </label>
        </div>
        <div
          className="d-flex justify-content-start form-group col-md-4"
        >
          <select
            className="formBox noerror form-control form-control-md"
            defaultValue="DEFAULT"
            name="occupancy"
            onChange={[Function]}
          >
            <option
              disabled={true}
              hidden={true}
              value="DEFAULT"
            >
              Select Option
            </option>
            <option
              value="Primary"
            >
              Primary Residence
            </option>
            <option
              value="Secondary"
            >
              Secondary Residence
            </option>
            <option
              value="Investment"
            >
              Investment
            </option>
          </select>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="justify-content-start form-group col-lg-8 col-md-8"
        />
        <div
          className="justify-content-start button form-group col-lg-4 col-md-4"
        >
          <button
            className="blackBackground btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="submit"
          >
            Quote Rates
          </button>
        </div>
      </div>
    </form>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="fade d-none alert alert-danger show"
          role="alert"
        >
          
        </div>
      </div>
    </div>
    <div
      className="d-flex row"
    >
      <p />
    </div>
  </div>
</div>
`;
